Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderTableView.cpp

```diff
@@ -31,16 +31,44 @@ namespace AzQtComponents
         if (index != rootIndex())
         {
             QAbstractItemView::setRootIndex(index);
-            emit rootIndexChanged(index);
+            emit tableRootIndexChanged(index);
         }
     }
 
-
     void AssetFolderTableView::SetShowSearchResultsMode(bool searchMode)
     {
         m_showSearchResultsMode = searchMode;
     }
 
+    void AssetFolderTableView::mousePressEvent(QMouseEvent* event)
+    {
+        const auto p = event->pos();
+        if (auto idx = indexAt(p); !idx.isValid())
+        {
+            selectionModel()->clear();
+            return;
+        }
+        TableView::mousePressEvent(event);
+    }
+
+    void AssetFolderTableView::mouseDoubleClickEvent(QMouseEvent* event)
+    {
+        const auto p = event->pos();
+        if (auto idx = indexAt(p); idx.isValid())
+        {
+            selectionModel()->select(idx, QItemSelectionModel::SelectionFlag::ClearAndSelect);
+            emit doubleClicked(idx);
+            return;
+        }
+    }
+
+    void AssetFolderTableView::contextMenuEvent(QContextMenuEvent* event)
+    {
+        const auto p = event->pos();
+        emit tableContextMenu(indexAt(p));
+        return;
```

Comment:
or this...

Reply:
again, sorry, I was unclear. I meant, you don't need a return statement at the bottom of a function.